r/src/compute-exec.cpp

```diff
@@ -146,20 +146,12 @@ std::shared_ptr<compute::ExecNode> ExecNode_Scan(
 
   options->dataset_schema = dataset->schema();
 
-  // ScanNode needs the filter to do predicate pushdown and skip partitions
-  options->filter = ValueOrStop(filter->Bind(*dataset->schema()));
```

Comment:
I'm not sure this is right.  I got rid of this to remove the call to `Bind()` but it looks like this chunk of code has more purposes than just that?

I was thinking it might just need the variable `filter` adding into `options`, but wasn't sure how to do that.  

I tried `options->filter = ValueOrStop(filter);` but got an error about `ValueOrStop` not being defined.
I also tried `options->filter = filter;` but get `no viable overloaded '='`.

Reply:
IIUC the ScanOptions needs the Expression but you have the shared_ptr so it should be this :

```
options->filter = filter.get();
```

Reply:
Perhaps:
```
options->filter = *filter
```